dynawo/sources/Models/Modelica/Dynawo/NonElectrical/Blocks/NonLinear/FirstOrderWithDelay.mo

```diff
@@ -0,0 +1,59 @@
+within Dynawo.NonElectrical.Blocks.NonLinear;
+
+/*
+* Copyright (c) 2015-2021, RTE (http://www.rte-france.com)
+* See AUTHORS.txt
+* All rights reserved.
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, you can obtain one at http://mozilla.org/MPL/2.0/.
+* SPDX-License-Identifier: MPL-2.0
+*
+* This file is part of Dynawo, an hybrid C++/Modelica open source suite of simulation tools for power systems.
+*/
+
+block FirstOrderWithDelay "Delay with a first order filter smoothing the input signal"
+
+  import Modelica;
+  import Dynawo;
+
+  extends Modelica.Blocks.Interfaces.SISO;
+
+  parameter Dynawo.Types.Time delayTime "Delay time of output with respect to input signal";
+  parameter Real y_start = 0 "Initial or guess value of output (= state)" annotation (
+    Dialog(group="Initialization"));
+
+  Modelica.Blocks.Nonlinear.FixedDelay fixedDelay(delayTime = delayTime) annotation(
+    Placement(visible = true, transformation(origin = {30, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+  Modelica.Blocks.Continuous.FirstOrder firstOrder(T = delayTime / 20, y_start = y_start) annotation(
+    Placement(visible = true, transformation(origin = {-30, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
+
+equation
+  connect(firstOrder.y, fixedDelay.u) annotation(
+    Line(points = {{-19, 0}, {18, 0}}, color = {0, 0, 127}));
+  connect(u, firstOrder.u) annotation(
+    Line(points = {{-120, 0}, {-42, 0}}, color = {0, 0, 127}));
+  connect(fixedDelay.y, y) annotation(
+    Line(points = {{42, 0}, {110, 0}}, color = {0, 0, 127}));
+
+  annotation (
+    Documentation(info= "<html><head></head><body><p>
+The Input signal is delayed by a given time instant, or more precisely:
+</p>
+<pre> y = u(time - delayTime) for time &gt; time.start + delayTime
+   = u(time.start)       for time ≤ time.start + delayTime
+</pre>
+<p>
+The FirstOrder block defines the transfer function between the input u
+and the input of the delay block as <em>first order</em> system:
+</p>
+<pre>                        k
+   fixedDelay.u = ------------ * u
```

Comment:
Could you add a preferred view for all new models ? Thanks

Reply:
Done